Userland/Libraries/LibGUI/DynamicWidgetContainer.cpp

```diff
@@ -0,0 +1,464 @@
+    m_detached_widgets_window.value()->show();
+    return {};
+}
+
+void DynamicWidgetContainer::close_all_detached_windows()
+{
+    for (auto window : DynamicWidgetContainer::s_open_windows.in_reverse())
+        window->close();
+}
+
+void DynamicWidgetContainer::register_open_window(NonnullRefPtr<GUI::Window> window)
+{
+    s_open_windows.append(window);
+}
+
+void DynamicWidgetContainer::unregister_open_window(NonnullRefPtr<GUI::Window> window)
+{
+    Optional<size_t> match = s_open_windows.find_first_index(window);
+    if (match.has_value())
+        s_open_windows.remove(match.value());
+}
+
+void DynamicWidgetContainer::handle_mouseup_event(MouseEvent& event)
+{
+    if (event.button() != MouseButton::Primary)
+        return;
+
+    if (m_is_dragging) {
+        // If we dropped the widget outside of ourself, we would like to detach it.
+        if (m_parent_container == nullptr && !rect().contains(event.position()))
+            set_view_state(ViewState::Detached);
+
+        if (m_parent_container != nullptr) {
+            bool should_move_position = m_parent_container->check_has_move_target(relative_position().translated(m_current_mouse_position), MoveTargetOperation::ClearAllTargets);
+
+            if (should_move_position)
+                m_parent_container->swap_widget_positions(*this, relative_position().translated(m_current_mouse_position));
+            else
+                set_view_state(ViewState::Detached);
+        }
+
+        m_is_dragging = false;
+
+        auto label_widget = m_controls_widget->find_descendant_of_type_named<GUI::LabelWithEventDispatcher>("section_label");
+        // Change the cursor back to normal after dragging is finished. Otherwise the curse will only change if the mouse moves.
+        label_widget->update_cursor(Gfx::StandardCursor::Arrow);
+
+        update();
+    }
+}
+
+void DynamicWidgetContainer::handle_mousemove_event(MouseEvent& event)
+{
+    auto changed_cursor = m_is_dragging ? Gfx::StandardCursor::Move : Gfx::StandardCursor::Arrow;
+    if (m_move_widget_knurl.contains(event.position()) && !m_is_dragging)
+        changed_cursor = Gfx::StandardCursor::Hand;
+
+    if (event.buttons() == MouseButton::Primary && !m_is_dragging) {
+        m_is_dragging = true;
+        m_drag_start_location = event.position();
+        changed_cursor = Gfx::StandardCursor::Move;
+    }
+
+    if (m_is_dragging) {
+        m_current_mouse_position = event.position();
+        if (m_parent_container != nullptr) {
+            m_parent_container->check_has_move_target(relative_position().translated(m_current_mouse_position), MoveTargetOperation::SetTarget);
+        }
+        update();
+    }
+
+    auto label_widget = m_controls_widget->find_descendant_of_type_named<GUI::LabelWithEventDispatcher>("section_label");
+    label_widget->update_cursor(changed_cursor);
```

Comment:
`DynamicWidgetContainerControls::get_event_dispatcher()`? Though, maybe that method should have a better name. (`get_section_label()`?)

It also feels a little weird to be reaching inside and modifying its child widgets instead of telling it to do so. But no big deal.